lib/find_non_repeated_letter.rb

```diff
@@ -9,6 +9,28 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
+  
+  array = str.chars
+  
+  final = array.length - 1
+  
+  answer = ""
+  
+  array.each_index do |i|
+    if i == 0 && array[i] != array[(i+1)]
+      answer = array[i]
+      break
+    elsif i == final && array[i] != array[(i-1)]
+      answer = array[i]
+      break
+    elsif array[i] != array[(i-1)] && array[i] != array[(i+1)]
```

Comment:
I'm not certain, but I don't think you need this part of the conditional: `array[i] != array[(i-1)]`. Try removing it and seeing how that changes things. See if you can explain why it might not be necessary, or–if I'm wrong about this–why it is needed.